python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -0,0 +1,301 @@
+                        "dict with parameters passed to Keras model fit method")
+
+    @keyword_only
+    def __init__(self, inputCol=None, outputCol=None, modelFile=None, imageLoader=None,
+                 outputMode="vector", optimizer=None, loss=None, kerasParams=None):
+        """
+        __init__(self, inputCol=None, outputCol=None, modelFile=None, imageLoader=None,
+                 outputMode="vector", optimizer=None, loss=None, kerasParams=None)
+        """
+        super(KerasImageFileEstimator, self).__init__()
+        kwargs = self._input_kwargs
+        self.setParams(**kwargs)
+
+    @keyword_only
+    def setParams(self, inputCol=None, outputCol=None, modelFile=None, imageLoader=None,
+                  outputMode="vector", optimizer=None, loss=None, kerasParams=None):
+        """
+        setParams(self, inputCol=None, outputCol=None, modelFile=None, imageLoader=None,
+                  outputMode="vector", optimizer=None, loss=None, kerasParams=None)
+        """
+        kwargs = self._input_kwargs
+        self._set(**kwargs)
+        return self
+
+    def setModelFile(self, value):
+        return self._set(modelFile=value)
+
+    def getModelFile(self):
+        return self.getOrDefault(self.modelFile)
+
+    def setOptimizer(self, value):
+        return self._set(optimizer=value)
+
+    def getOptimizer(self):
+        return self.getOrDefault(self.optimizer)
+
+    def setLoss(self, value):
+        return self._set(loss=value)
+
+    def getLoss(self):
+        return self.getOrDefault(self.loss)
+    
+    def _validateParams(self):
+        """
+        Check Param values so we can throw errors on the driver, rather than workers.
+        """
+        if not self.isDefined(self.optimizer):
+            raise ValueError("Required Param optimizer must be specified but was not.")
+        if not keras_utils.is_valid_optimizer(self.optimizer):
+            raise ValueError("Named optimizer not supported in Keras: %s" % self.getOptimizer)
+        if not self.isDefined(self.loss):
+            raise ValueError("Required Param loss must be specified but was not.")
+        if not keras_utils.is_valid_loss_function(self.loss):
+            raise ValueError("Named loss not supported in Keras: %s" % self.getLoss)
+        # TODO: check inputCol
+
+    def _localFit(self, featuresBc, labelsBc, modelBytesBc, paramMap):
+        # Copy the estimator to add to the Params without modifying this instance.
+        estimator = self.copy(paramMap)
+
+        features = featuresBc.value()
+        labels = None if labelsBc is None else labelsBc.value()
+        model = keras_utils.bytes_to_model(modelBytesBc.value())
+
+        model.compile(optimizer=estimator.getOptimizer(), loss=estimator.getLoss())
+        kerasParams = estimator.getKerasParams()
+        model.fit(x=features, y=labels, **kerasParams)
+        return model
+
+    def _getNumpyFeaturesAndLabels(self, dataset):
```

Comment:
Features might fail to load due to various resource acquisition errors. 
Thus we must get features and their labels together.